src/test/java/com/iexec/core/task/TaskRepositoryTest.java

```diff
@@ -0,0 +1,111 @@
+    @Container
+    private static final MongoDBContainer mongoDBContainer = new MongoDBContainer(DockerImageName.parse("mongo:4.2"));
+
+    @DynamicPropertySource
+    static void registerProperties(DynamicPropertyRegistry registry) {
+        registry.add("spring.data.mongodb.uri", mongoDBContainer::getReplicaSetUrl);
+    }
+
+    @Autowired
+    private TaskRepository taskRepository;
+
+    @BeforeEach
+    void init() {
+        taskRepository.deleteAll();
+    }
+
+    private String generateChainId() {
+        int length = 64;
+        StringBuilder sb = new StringBuilder("0x");
+        for (int j = 0; j < length; j++) {
+            sb.append(HEX_ARRAY[generator.nextInt(HEX_ARRAY.length)]);
+        }
+        return sb.toString();
+    }
+
+    private List<Task> queryTasksOrderedByStatusThenContributionDeadline() {
+        return taskRepository.findByCurrentStatus(Arrays.asList(INITIALIZED, RUNNING),
+                Sort.by(Sort.Order.desc(Task.CURRENT_STATUS_FIELD_NAME),
+                        Sort.Order.asc(Task.CONTRIBUTION_DEADLINE_FIELD_NAME))
+        );
+    }
+
+    @Test
+    void shouldFailWithDuplicateUniqueDealIdx() {
+        Task task1 = getStubTask(maxExecutionTime);
+        Task task2 = getStubTask(maxExecutionTime);
+        Assertions.assertThatThrownBy(() -> taskRepository.saveAll(Arrays.asList(task1, task2)))
+                .isInstanceOf(DuplicateKeyException.class)
+                .hasCauseExactlyInstanceOf(com.mongodb.MongoBulkWriteException.class)
+                .hasMessageContaining("duplicate key error collection: iexec.task index: unique_deal_idx dup key");
+    }
+
+    @Test
+    void shouldFailWithDuplicateChainTaskId() {
+        Task task1 = getStubTask(maxExecutionTime);
+        task1.setTaskIndex(0);
+        Task task2 = getStubTask(maxExecutionTime);
+        task2.setTaskIndex(1);
+        Assertions.assertThatThrownBy(() -> taskRepository.saveAll(Arrays.asList(task1, task2)))
+                .isInstanceOf(DuplicateKeyException.class)
+                .hasCauseExactlyInstanceOf(com.mongodb.MongoBulkWriteException.class)
+                .hasMessageContaining("duplicate key error collection: iexec.task index: chainTaskId dup key");
+    }
+
+    @Test
+    void shouldFindTasksOrderedByCurrentStatusAndContributionDeadline() {
+        List<Task> tasks = new ArrayList<>();
+        for (int i = 0; i < 10; i++) {
+            Task task = getStubTask(maxExecutionTime);
+            task.setChainDealId(generateChainId());
+            task.setChainTaskId(generateChainId());
+            task.setCurrentStatus(generator.nextInt(50) % 2 == 0 ? RUNNING : INITIALIZED);
+            int amountToAdd = generator.nextInt(10);
+            task.setContributionDeadline(Date.from(Instant.now().plus(amountToAdd, ChronoUnit.MINUTES)));
+            tasks.add(task);
+        }
+        taskRepository.saveAll(tasks);
+        tasks.sort(Comparator.comparing(Task::getCurrentStatus, Comparator.reverseOrder())
```

Comment:
I think it is interesting/clearer to add at least 1 test where expected task list is explicitly defined.

Reply:
Would it be alright to have a simple test with fixed data and a fuzzy one with randomization to try different combinations ?

Reply:
That would be perfect, having both benefits!